### PR TITLE
Fix generate connecting to Kafka REST proxy

### LINKED ISSUE #37: Generate should not require connection to Kafka REST Proxy
recent regression introduced by #3. `generate` step should run without connecting to Kafka REST or other services

### PR DESCRIPTION
**Pull request by disrupted**
Fixes #37

### CODE REVIEW

**Comment by philipp94831** [comment]
It works as expected again :-)

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please add a test that the proxy wrapper is not initialized during generate

**Comment by disrupted** [comment]
> Please add a test that the proxy wrapper is not initialized during generate

the proxy wrapper is always initialized as part of the TopicHandler. it just won't have to connect to Kafka REST during `generate`

**Comment by disrupted** [comment]
I'll see what I can do about a test :)

**Comment by disrupted** [comment]
this should now be covered by `test_pipeline.py`. Also `kafka_rest_host` is no longer set

**Comment by philipp94831** [comment]
> this should now be covered by `test_pipeline.py`. Also `kafka_rest_host` is no longer set

I don't quite see how that is covered. There has nothing been running on the port anyways so there was no connection attempt beforehand, right?

**Comment by disrupted** [comment]
There was something else missing. Previously the handlers were mocked. I've removed this mock now, so that the tests in `test_pipeline.py` (which all run the `generate` command) would fail if a connection to Kafka REST proxy is attempted.

**Review by philipp94831 — APPROVED** [review]

**Comment by raminqaf** [comment]
But this PR does not prevent calling `helm repo add` and `helm repo update` when calling `generate` right?

**Review by raminqaf — CHANGES_REQUESTED** [review]
Generally, we should avoid initializing the Handlers when calling the `generate` command. They are not used during the generation process. This is a quick fix but still not a good solution.

**Comment by philipp94831** [comment]
> Generally, we should avoid initializing the Handlers when calling the `generate` command. They are not used during the generation process. This is a quick fix but still not a good solution.

I agree. However, I think it makes sense to do this in a follow-up PR in order to get rid of the first issue (Kafka REST connectivity in generate) already. Fine for you?

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 .DS_Store
 .envrc
 .vscode
+pipelines/
+defaults/
```
Comment (raminqaf) [inline .gitignore:7]: why did you add these?

**Review by raminqaf — APPROVED** [review]

**Comment by philipp94831** [comment]
I created #42

File: .gitignore (hunk 1 above)
Comment (disrupted) [reply .gitignore:7]: I usually keep these around for testing. felt a bit anxious to accidentally commit some of it, so I added it to ignore to be safe

**Comment by disrupted** [comment]
> But this PR does not prevent calling `helm repo add` and `helm repo update` when calling `generate` right?

no, this is only targeting #37
